assets/js/custom/Translation.js

```diff
@@ -46,95 +26,65 @@ function Translation (translatedByLine) {
       type: 'get',
       success: function (data) {
-        displayLanguageMap = data
+        self.displayLanguageMap = data
       }
     })
   }
 
-  function splitTranslatedByLine () {
+  splitTranslatedByLine () {
+    const self = this
     let firstLanguage = '%sourceLanguage%'
     let secondLanguage = '%targetLanguage%'
-    if (!isSourceLanguageFirst()) {
+    if (!this.isSourceLanguageFirst()) {
       firstLanguage = '%targetLanguage%'
       secondLanguage = '%sourceLanguage%'
     }
 
-    translatedByLineMap = {
-      before: translatedByLine.substring(0, translatedByLine.indexOf(firstLanguage)),
-      between: translatedByLine.substring(translatedByLine.indexOf(firstLanguage) + firstLanguage.length, translatedByLine.indexOf(secondLanguage)),
-      after: translatedByLine.substring(translatedByLine.indexOf(secondLanguage) + secondLanguage.length)
+    self.translatedByLineMap = {
+      before: self.translatedByLine.substring(0, self.translatedByLine.indexOf(firstLanguage)),
+      between: self.translatedByLine.substring(self.translatedByLine.indexOf(firstLanguage) + firstLanguage.length, self.translatedByLine.indexOf(secondLanguage)),
+      after: self.translatedByLine.substring(self.translatedByLine.indexOf(secondLanguage) + secondLanguage.length)
     }
   }
 
-  function isTranslationNotAvailable (commentId) {
-    return $('#comment-text-translation-' + commentId).attr('lang') !== targetLanguage
+  isSourceLanguageFirst () {
+    return this.translatedByLine.indexOf('%sourceLanguage%') < this.translatedByLine.indexOf('%targetLanguage%')
   }
 
-  function isSourceLanguageFirst () {
-    return translatedByLine.indexOf('%sourceLanguage%') < translatedByLine.indexOf('%targetLanguage%')
+  isTranslationNotAvailable (elementId) {
+    return $(elementId).attr('lang') !== this.targetLanguage
   }
 
-  function openGoogleTranslatePage (commentId) {
-    const text = document.getElementById('comment-text-' + commentId).innerText
+  openGoogleTranslatePage (text) {
     window.open(
-      'https://translate.google.com/?q=' + encodeURIComponent(text) + '&sl=auto&tl=' + targetLanguage,
+      'https://translate.google.com/?q=' + encodeURIComponent(text) + '&sl=auto&tl=' + this.targetLanguage,
       '_self'
     )
   }
 
-  function setTranslatedCommentData (commentId, data) {
-    $('#comment-text-translation-' + commentId).text(data.translation)
-    $('#comment-text-translation-' + commentId).attr('lang', data.target_language)
-
-    $('#comment-translation-before-languages-' + commentId).text(translatedByLineMap.before.replace('%provider%', providerMap[data.provider]))
-    $('#comment-translation-between-languages-' + commentId).text(translatedByLineMap.between.replace('%provider%', providerMap[data.provider]))
-    $('#comment-translation-after-languages-' + commentId).text(translatedByLineMap.after.replace('%provider%', providerMap[data.provider]))
+  setTranslationCredit (creditContainers, data) {

```

Comment:
I think this is a good idea if we are planning to implement the selectors later on. However, if we aren't going to do the language selectors, I can simplify the logic so that the ordering of the source and target languages doesn't have to be figured out and then this class would only download the languages map and then set the byline which I think would make more sense in the Translation class